packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts

```diff
@@ -47,82 +50,96 @@ export class StackdriverStatsExporter implements StatsEventListener {
     this.projectId = options.projectId;
     this.metricPrefix =
         options.prefix || StackdriverStatsExporter.CUSTOM_OPENCENSUS_DOMAIN;
+    this.displayNamePrefix =
+        options.prefix || StackdriverStatsExporter.DEFAULT_DISPLAY_NAME_PREFIX;
     this.logger = options.logger || logger.logger();
-    this.viewToUpload = {};
+    this.onMetricUploadError = options.onMetricUploadError;
+  }
 
+  /**
+   * Creates a Stackdriver Stats exporter with a StackdriverExporterOptions.
+   */
+  start(): void {
     this.timer = setInterval(async () => {
       try {
-        await this.uploadViews();
+        await this.export();
       } catch (err) {
-        if (typeof options.onMetricUploadError === 'function') {
-          options.onMetricUploadError(err);
+        if (typeof this.onMetricUploadError === 'function') {
+          this.onMetricUploadError(err);
         }
       }
     }, this.period);
   }
 
   /**
-   * Is called whenever a view is registered.
-   * @param view The registered view.
+   * Polls MetricProducerManager from Metrics library for all registered
+   * MetricDescriptors, and upload them as TimeSeries to StackDriver.
    */
-  onRegisterView(view: View) {
-    return this.authorize().then((authClient) => {
-      const request = {
-        name: `projects/${this.projectId}`,
-        resource: this.createMetricDescriptorData(view),
-        auth: authClient
-      };
+  async export() {
+    const metricsList: Metric[] = [];
+    const metricProducerManager: MetricProducerManager =
+        Metrics.getMetricProducerManager();
+    for (const metricProducer of metricProducerManager.getAllMetricProducer()) {
+      for (const metric of metricProducer.getMetrics()) {
+        const isRegistered =
+            await this.registerMetricDescriptor(metric.descriptor);
```

Comment:
Would it be worth running these `registerMetricDescriptor` calls in parallel? Not sure how useful it would be but they don't seem dependent on each other in this loop.

It seems like if you ran them all in parallel with something like `Promise.all` then you could aggregate their results together into the `metricsList` list once they all complete.

Reply:
The idea is to register each metricDescriptor once (and ignore metricDescriptor that are already registered). If we run  ```registerMetricDescriptor``` in parallel, there are chances will make more external calls than usual. WDYT?

Reply:
OK. I suppose registering metric descriptors may not happen that often, so it's probably OK to leave as is. Maybe just something we can think of if anybody has a performance issue with it that we could try to compress it to a `Set` and run in parallel, but I like leaving it alone for now.

Reply:
This is certainly a good point, I have added TODO comment for now to track this in future.